### PR TITLE
MAINT,TST: Refactor F2PY testsuite

### PR DESCRIPTION
**Pull request by HaoZeke**
- **Closes** #19922. 
- **Requires** #20479.
    -  Not only are some helpers added there, but also the formatting changes to appease `flake8` are in that PR
    - Once it is merged this becomes far simpler to review (one commit) 

Essentially this makes better usage of the `util.py` helpers. It will also make testing more transparent and easier to extend since now one can directly write nice Fortran formatted tests instead of massaging lines in `python`.

Additionally, one point which came up was the speed. AFAIK the time taken to run the tests has *decreased*, and not increased.

```bash
# On main
python runtests.py -m full -s f2py
# 670 passed, 6 skipped in 66.84s (0:01:06)
# On this branch
python runtests.py -m full -s f2py
# 670 passed, 6 skipped in 60.85s (0:01:00)
```

Similar behavior in terms of time taken was observed in the [skeletal test repository](https://github.com/HaoZeke/f2py_skel/pull/6).

Other changes:
- Almost no dependencies left on `numpy.testing` other than `IS_PYPY`

### CODE REVIEW

**Review by pearu — APPROVED** [review]
LGTM!

File: numpy/f2py/tests/test_abstract_interface.py [hunk 1]
```diff
@@ -3,65 +3,22 @@
 from . import util
 from numpy.f2py import crackfortran
 
+from numpy.testing import assert_
```
Comment (melissawm) [inline numpy/f2py/tests/test_abstract_interface.py:6]: Minor nit: at some point there was a discussion about removing uses of `assert_` to reduce dependency on `numpy.testing`

Comment (HaoZeke) [reply numpy/f2py/tests/test_abstract_interface.py:6]: Excellent point, given the discussion here: https://github.com/scipy/scipy/issues/8583

I agree that we can do without `assert_`, will kill it.

**Comment by melissawm** [comment]
This looks good - thanks @HaoZeke

**Pull request merged**